File: python/transit.py
"""
Human Design API — Transit Calculation Example
https://humandesignhub.app/human-design-api
"""
import requests
from datetime import datetime, timezone

API_BASE = "https://api.humandesignhub.app/v1"
API_KEY = "your_api_key_here"


def get_current_transit():
    """Get the current global transit chart."""
    now = datetime.now(timezone.utc).strftime("%Y-%m-%dT%H:%M+00:00")
    response = requests.post(
        f"{API_BASE}/transit",
        headers={
            "Content-Type": "application/json",
            "X-API-Key": API_KEY,
        },
        json={"datetime": now},
    )
    response.raise_for_status()
    return response.json()


def get_personal_transit(birth_datetime: str):
    """Get a personal transit overlay (birth chart + current transit)."""
    now = datetime.now(timezone.utc).strftime("%Y-%m-%dT%H:%M+00:00")
    response = requests.post(
        f"{API_BASE}/transit-overlay",
        headers={
            "Content-Type": "application/json",
            "X-API-Key": API_KEY,
        },
        json={
            "birth_datetime": birth_datetime,
            "transit_datetime": now,
        },
    )
    response.raise_for_status()
    return response.json()


if __name__ == "__main__":
    print("=== Current Global Transit ===")
    transit = get_current_transit()
    print(f"Transit Gates: {len(transit.get('gates', []))} active")

    print("\n=== Personal Transit Overlay ===")
    birth = "1990-01-01T12:34+09:00"
    overlay = get_personal_transit(birth)
    print(f"Electromagnetics: {overlay.get('electromagnetics')}")
    print(f"Channel Completions: {overlay.get('channel_completions')}")

File: javascript/transit.js
/**
 * Human Design API — Transit Calculation Example
 * https://humandesignhub.app/human-design-api
 */

const API_BASE = "https://api.humandesignhub.app/v1";
const API_KEY = "your_api_key_here";

async function getCurrentTransit() {
  const now = new Date().toISOString().slice(0, 16) + "+00:00";
  const response = await fetch(`${API_BASE}/transit`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      "X-API-Key": API_KEY,
    },
    body: JSON.stringify({ datetime: now }),
  });

  if (!response.ok) {
    throw new Error(`API error: ${response.status}`);
  }

  return response.json();
}

async function getPersonalTransit(birthDatetime) {
  const now = new Date().toISOString().slice(0, 16) + "+00:00";
  const response = await fetch(`${API_BASE}/transit-overlay`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      "X-API-Key": API_KEY,
    },
    body: JSON.stringify({
      birth_datetime: birthDatetime,
      transit_datetime: now,
    }),
  });

  if (!response.ok) {
    throw new Error(`API error: ${response.status}`);
  }

  return response.json();
}

console.log("=== Current Global Transit ===");
const transit = await getCurrentTransit();
console.log(`Transit Gates: ${transit.gates?.length ?? 0} active`);

console.log("\n=== Personal Transit Overlay ===");
const birth = "1990-01-01T12:34+09:00";
const overlay = await getPersonalTransit(birth);
console.log(`Electromagnetics: ${JSON.stringify(overlay.electromagnetics)}`);
console.log(`Channel Completions: ${JSON.stringify(overlay.channel_completions)}`);
